File: src/main/kotlin/com/huawei/cangjie/diagnostics/rendering/SimpleDiagnosticRenderer.java
package com.huawei.cangjie.diagnostics.rendering;

import com.huawei.cangjie.diagnostics.Diagnostic;
import org.jetbrains.annotations.NotNull;

public class SimpleDiagnosticRenderer implements DiagnosticRenderer<Diagnostic> {
    private final String message;

    public SimpleDiagnosticRenderer(@NotNull String message) {
        this.message = message;
    }

    @NotNull
    @Override
    public String render(@NotNull Diagnostic diagnostic) {
        return message;
    }

    @NotNull
    @Override
    public Object[] renderParameters(@NotNull Diagnostic diagnostic) {
        return new Object[0];
    }
}

File: src/main/kotlin/com/huawei/cangjie/diagnostics/rendering/SimpleDiagnosticRenderer.kt
package com.huawei.cangjie.diagnostics.rendering

import com.huawei.cangjie.diagnostics.Diagnostic
import java.text.MessageFormat

class SimpleDiagnosticRenderer(private val message: String) :
    DiagnosticRenderer<Diagnostic > {
    override fun render(diagnostic: Diagnostic): String {

        val str = CangJieDiagnosisBundle.rawMessage(diagnostic.factory.name)
        if(str == "!${diagnostic.factory.name}!"){
            return message
        }

        return str
    }

    override fun renderParameters(diagnostic: Diagnostic): Array<Any?> {
        return arrayOfNulls(0)
    }
}
